swiftspec/core.py

```diff
@@ -139,21 +139,52 @@ async def set_session(self):
         return self._session
 
     def get_tokens_from_env(self):
+        auth = []
         token = os.environ.get("OS_AUTH_TOKEN")
         url = os.environ.get("OS_STORAGE_URL")
         if token and url:
-            return [{"token": token, "url": url}]
-        else:
-            return []
+            auth.append({"token": token, "url": url})
+
+        sig = os.environ.get("TEMP_URL_SIG")
+        expire = os.environ.get("TEMP_URL_EXPIRES")
+        prf = os.environ.get("TEMP_URL_PREFIX")
+
+        if url and sig and expire:
+            if prf:
+                auth.append(
+                    {
+                        "url": url,
+                        "temp_url_sig": sig,
+                        "temp_url_expires": expire,
+                        "temp_url_prefix": prf,
+                    }
+                )
+            else:
+                auth.append(
+                    {"url": url, "temp_url_sig": sig, "temp_url_expires": expire}
+                )
+        return auth
 
     def headers_for_url(self, url):
         headers = {}
         for auth in self.auth:
-            if url.startswith(auth["url"]):
+            if url.startswith(auth["url"]) and "token" in auth:
```

Comment:
This change probably shouldn't be there in this form. The loop over all the `auth` methods should stop as soon as there is a match on the `url`. It might then be the case, that for the matched auth method, the `headers` should be empty (because it's `params`-based).

(it's the same issue for the loop in `params_for_url`)

Reply:
I think if you have a url AND a token, that should always be preferred, right? however I do not understand why token should not be checked.

Reply:
@d70-t i do not understand how to decide what to take unless using the check for token

Reply:
My thinking was that `auth` would be a list of authentication methods and the rule would be to apply the first method which matches on the requested url. Along this line, checking for `token` would be a way of skipping the first entry matching the `url` (but we don't want to skip).

At some point, I thought if `auth` might be better implemented as a `dict`, but that doesn't support prefix-matching, thus it became a `list`.

Reply:
> @d70-t i do not understand how to decide what to take unless using the check for token

It would be up to the method creating the `auth` list, to ensure only the right option (token or temporary) is part of the list,